### PR TITLE
Updated RPC code.

### PR DESCRIPTION
**Pull request by TimSimpsonR**
- Created a OS common config facade.
- Added package reddwarf.openstack.common.rpc as well as excutils, jsonutils and some other files. Modified it a bit to use the config facade.
- Deleted reddwarf.rpc.
- Moved _wrapper code to count running tasks out of reddwarf/taskmanager/manager.py and put it into the common service Manager base class.

### CODE REVIEW

File: reddwarf/openstack/common/rpc/amqp.py [hunk 1]
```diff
@@ -229,27 +252,29 @@ def __call__(self, message_data):
...
+        the old behavior of magically calling the specified method on the
+        proxy we have here.
+        """
+        ctxt.update_store()
         try:
-            node_func = getattr(self.proxy, str(method))
-            node_args = dict((str(k), v) for k, v in args.iteritems())
-            # NOTE(vish): magic is fun!
-            rval = node_func(context=ctxt, **node_args)
+            rval = self.proxy.dispatch(ctxt, version, method, **args)
```
Comment (ed-) [inline reddwarf/openstack/common/rpc/amqp.py:277]: My eye twitches at the sight of `**args` but I see what you're doing there.

**Comment by ed-** [comment]
+1

**Comment by imsplitbit** [comment]
+1

**Comment by hub-cap** [comment]
Closing so cp16net can reissue

**Pull request closed without merging**